static/extensions/PuzzlingGGG/underspeech.js

```diff
@@ -0,0 +1,196 @@
+class Underspeech {
+  constructor() {
+    this.dialogueContainer = this.createDialogueContainer();
+    const canvas = document.getElementById('sulfurous') || document.querySelector('canvas');
+    if (canvas && canvas.parentNode) {
+      canvas.parentNode.insertBefore(this.dialogueContainer, canvas.nextSibling);
+    }
+    this.dialogueText = '';
+    this.timeout = null;
+}
+
+    getInfo() {
+        return {
+            id: 'puzzlingunderspeech',
+            name: 'Underspeech',
+            blocks: [
+                {
+                    opcode: 'showDialogueFromJSON',
+                    blockType: Scratch.BlockType.COMMAND,
+                    text: 'say [JSON] by [SPEAKER] as [TYPE]',
+                    arguments: {
+                        JSON: {
+                            type: Scratch.ArgumentType.STRING,
+                            defaultValue: '{"line": [{"text": "Hello, ", "style": {"font": "Arial", "size": "14px", "color": "black"}}]}'
+                        },
+                        SPEAKER: {
+                            type: Scratch.ArgumentType.STRING,
+                            defaultValue: "Dango"
+                        },
+                        TYPE: {
+                            type: Scratch.ArgumentType.STRING,
+                            menu: 'typeMenu',
+                            defaultValue: 'normal'
+                        },
+                    }
+                },
+                {
+                    opcode: 'hideDialogue',
+                    blockType: Scratch.BlockType.COMMAND,
+                    text: 'hide dialogue'
+                },
+                '---','---','---',
```

Comment:
you dont need 3 '---'

It does/adds nothing

Reply:
erm actually nuhuh it adds three spaces to the category xml :trollface:

Reply:
> erm actually nuhuh it adds three spaces to the category xml :trollface:

only 1

Reply:
> > erm actually nuhuh it adds three spaces to the category xml :trollface:
> 
> only 1

notice my reference to "category XML" 💀
you can only *see* one, there are still others added to the xml though